app/main/views/frameworks.py

```diff
@@ -212,32 +212,40 @@ def get_framework_suppliers(framework_slug):
         Framework.slug == framework_slug
     ).first_or_404()
 
-    # TODO: Work out a way to order by current_framework_agreement.signed_agreement_returned_at
-    # Listing agreements is something done for Admin only (suppliers only retrieve their individual agreements)
-    # CCS always want to work from the oldest returned date to newest, so order by ascending date
+    # we're going to need an alias for the current_framework_agreement join
+    cfa = orm.aliased(SupplierFramework._CurrentFrameworkAgreement)
+    supplier_frameworks = SupplierFramework.query.outerjoin(
+        cfa, SupplierFramework.current_framework_agreement
+    ).options(
+        orm.contains_eager(
+            SupplierFramework.current_framework_agreement, alias=cfa
+        )
+    )
 
-    # TODO: Work out a way to query.filter() on current_framework_agreement rather than manipulating with Python
-    supplier_frameworks = SupplierFramework.query.filter(
+    supplier_frameworks = supplier_frameworks.filter(
         SupplierFramework.framework_id == framework.id
+    ).order_by(
+        cfa.most_recent_signature_time.asc().nullsfirst(),
```

Comment:
That's a point. I had just considered it a spec to follow and didn't question it. Come to think of it if ordered by `most_recent_signature_time` it will also screw around with the "next" order...